src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py

```diff
@@ -406,48 +406,36 @@ def process_file_upload_batch_parameters(namespace):
     # 2. try to extract account name and container name from destination string
     from .storage_url_helpers import StorageResourceIdentifier
     identifier = StorageResourceIdentifier(namespace.destination)
-    if not identifier.is_url():
-        namespace.destination_share_name = namespace.destination
-    elif identifier.filename or identifier.directory:
-        raise ValueError('incorrect usage: destination must be a file share url')
-    else:
-        namespace.destination_share_name = identifier.share
+    if identifier.is_url():
+        if identifier.filename or identifier.directory:
+            raise ValueError('incorrect usage: destination must be a file share url')
 
-    if not namespace.account_name:
-        namespace.account_name = identifier.account_name
+        namespace.destination = identifier.share
```

Comment:
Okay, wanted to make sure the functionality is still present. I still think everything (account name, share, directory path, and pattern) should be able to be parsed from the URL, but as long as the functionality is present we can check this in.